### PR TITLE
Change --insecure flag to --self-signed

### LINKED ISSUE #2542: Change --insecure flag to --skipcertcheck
*** Not using standard template ***

`mc` currently uses `--insecureflag` flag for the case where self signed certificates are used. The name is a little misleading. So, this needs to be changed to `--skipcertcheck`

### PR DESCRIPTION
**Pull request by Praveenrajmani**
Will still support `--insecure` as a legacy option. And continue supporting `--self-signed` from now

<b>Tests</b> - Both `./mc admin --self-signed info play` and `./mc admin --insecure info play` will work

Fixes #2542

### CODE REVIEW

File: cmd/flags.go [hunk 1]
```diff
@@ -52,6 +52,10 @@ var globalFlags = []cli.Flag{
 	},
 	cli.BoolFlag{
 		Name:  "insecure",
+		Usage: " (legacy) Disable SSL certificate verification.",
```
Comment (harshavardhana) [inline cmd/flags.go]: We should mention that skipcert should be used..

File: cmd/session-migrate.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ import (
 )
 
 // Migrates session header version '7' to '8'. The only
-// change was the adding of insecure global flag
+// change was the adding of skipCertCheck global flag
 func migrateSessionV7ToV8() {
```
Comment (vadmeste) [inline cmd/session-migrate.go]: I think you need to create a new session version V9 and migrate from V8 to V9 by renaming insecure to skipCertCheck

Comment (harshavardhana) [reply cmd/session-migrate.go]: yes you have to @Praveenrajmani

Comment (Praveenrajmani) [reply cmd/session-migrate.go]: Done

File: cmd/flags.go (hunk 1 above)
Comment (Praveenrajmani) [reply cmd/flags.go]: Done

**Comment by Praveenrajmani** [comment]
The changes are done and ready for re-review

File: cmd/session-migrate.go [hunk 3]
```diff
@@ -25,8 +25,63 @@ import (
...
+		sessionVersion, e := strconv.Atoi(sV8.Header.Version)
+		fatalIf(probe.NewError(e), "Unable to load version `8`. Migration failed please report this issue at https://github.com/minio/mc/issues.")
+		if sessionVersion > 8 { // It is new format.
+			continue
+		}
+
+		sessionFile, err := getSessionFile(sid)
+		fatalIf(err.Trace(sid), "Unable to get session file.")
+
+		// Initialize v8 header and migrate to new config.
```
Comment (nitisht) [inline cmd/session-migrate.go]: update the comment here

File: cmd/session-migrate.go [hunk 4]
```diff
@@ -25,8 +25,63 @@ import (
...
+		e = qs.Save(sessionFile)
+		fatalIf(probe.NewError(e).Trace(sid, sessionFile), "Unable to migrate session from '8' to '9'.")
+
+		console.Println("Successfully migrated `" + sessionFile + "` from version `" + sV8.Header.Version + "` to " + "`" + sV9Header.Version + "`.")
+	}
+}
+
 // Migrates session header version '7' to '8'. The only
-// change was the adding of insecure global flag
+// change was the adding of skipCertCheck global flag
```
Comment (nitisht) [inline cmd/session-migrate.go]: This comment can be left unchanged.

Comment (Praveenrajmani) [reply cmd/session-migrate.go]: Done

File: cmd/session-migrate.go (hunk 3 above)
Comment (Praveenrajmani) [reply cmd/session-migrate.go]: Done

**Comment by Praveenrajmani** [comment]
Is this PR still blocked ? @deekoder

**Comment by deekoder** [comment]
We are still finalizing the flag name @Praveenrajmani

**Comment by kannappanr** [comment]
@Praveenrajmani We have finalized the name as --self-signed. Please look at #2542 for more info on this part. There is also another suggestion from @aead We can probably do that as part of another PR.

**Comment by Praveenrajmani** [comment]
> @Praveenrajmani We have finalized the name as --self-signed. Please look at #2542 for more info on this part. We need to have some more clarification about how this flag should behave

Changed the flag-name as `self-signed`. Ready for the re-review.
Yes lets do that in a separate PR after the clarification

**Comment by harshavardhana** [comment]
> Changed the flag-name as `self-signed`. Ready for the re-review.
> Yes lets do that in a separate PR after the clarification

Fix the commit title and message

File: cmd/flags.go [hunk 5]
```diff
@@ -52,6 +52,10 @@ var globalFlags = []cli.Flag{
 	},
 	cli.BoolFlag{
 		Name:  "insecure",
+		Usage: " (deprecated) Use skipcertcheck instead",
```
Comment (harshavardhana) [inline cmd/flags.go]: Should say `--self-signed`

File: docs/minio-client-complete-guide.md [hunk 6]
```diff
@@ -238,7 +238,7 @@ Quiet option suppress chatty console output.
 ### Option [--config-folder]
 Use this option to set a custom config path.
 
-### Option [ --insecure]
+### Option [ --skipccertcheck]
```
Comment (harshavardhana) [inline docs/minio-client-complete-guide.md]: should be --self-signed

File: cmd/flags.go (hunk 5 above)
Comment (Praveenrajmani) [reply cmd/flags.go]: Done

File: docs/minio-client-complete-guide.md (hunk 6 above)
Comment (Praveenrajmani) [reply docs/minio-client-complete-guide.md]: Done

**Comment by deekoder** [comment]
Can you resolve the conflicts?

**Comment by kannappanr** [comment]
Closing this PR as this will not be used. We will be implementing this in a different way as captured by https://github.com/minio/mc/issues/2596 and https://github.com/minio/minio/issues/6801

**Pull request closed without merging**